piff/basis_interp.py

```diff
@@ -446,13 +492,24 @@ def __init__(self, order, keys=('u','v'), max_order=None, use_qr=False, use_jax=
             self._max_order = np.max(self._orders)
         else:
             self._max_order = max_order
-        self.use_qr = use_qr
 
-        self.use_jax = use_jax
+        self.solver = solver
+
+        valid_solver = ["scipy", "qr", "jax", "cpp", "cpp32"]
+
+        if solver not in valid_solver:
+            raise ValueError(f"{solver} is not a valid solver. Valid solver are {valid_solver}")
+
+        # To match old API when jax and cpp were not part of solving
+        # basis interp.
+        if use_qr and solver not in ["scipy", "qr"]:
+            raise NotImplementedError(f"use_qr and {solver} are not compatible")
+        if use_qr:
+            self.solver = "qr"
```

Comment:
We should probably deprecate the `use_qr` option.  (I don't think anyone uses it anyway, but better to maintain backwards compatibility.)  So add a warning that `use_qr=True` should now be written `solver='qr'`.